feature/franka_shutdown

v8p1197 opened the pull request:
This PR creates a new script to shut down Franka programmatically.

v8p1197 reviewed (commented):
This PR creates a new script to shut down Franka programmatically.

jk-ethz commented:
Thanks for opening the PR. I will check this with our robot soon.

christian-rauch commented:
This works as expected on a Franka Panda (a.k.a. FER1):
```
Logging in...
Successfully logged in.
Requesting a control token...
Received control token is ******************************************** with id **********.
Successfully acquired control over the robot.
Shutting down...
The robot is shutting down. Please wait for the yellow lights to turn off, then switch the control box off.
```

@jk-ethz Can you merge this?

v8p1197 commented:
> This works as expected on a Franka Panda (a.k.a. FER1):

Thanks @christian-rauch for your feedback :) glad to help

christian-rauch commented:
@jk-ethz Generally, is this repo still maintained? Or should we simply fork it and maintain fixes and new features there?

jk-ethz commented:
@christian-rauch Yes, we still try to add features if they are well tested and do not break existing functionality.
I will take a look at your PR soon.

jk-ethz commented on franka_client.py:
```diff
@@ -0,0 +1,85 @@
+        active_token_id = self._get_active_token_id()
+        return active_token_id is None or active_token_id == self._token_id
+
+    def _request_token(self, physically=False):
```
Can we also include the `_release_token` here as the active counterpart?

v8p1197 replied:
Done.

v8p1197 commented:
Hi @jk-ethz, thank you so much for your review.
I have updated the PR according to your comment.
I have also updated the documentation to add instructions to run the new script.

christian-rauch commented:
I am getting a `ModuleNotFoundError: No module named 'franka_client'` with this. The `setup.py` is only installing the two scripts, but it is not installing the `franka_client.py`. And there seems to be an `__init__.py` missing. I guess everything should be in the `franka_lock_unlock` folder and installed at once.

v8p1197 commented:
@christian-rauch OK, I will check in these days and commit the fix :)

christian-rauch commented:
> @christian-rauch OK, I will check in these days and commit the fix :)

I went ahead and fixed this plus some additional issues in https://github.com/jk-ethz/franka_lock_unlock/pull/8.

v8p1197 commented:
> I went ahead and fixed this plus some additional issues in #8.

@christian-rauch Thank you! Sorry I didn't have time to work with the arm in these days.
@jk-ethz maybe you can close this PR and check #8.

jk-ethz commented:
> @jk-ethz maybe you can close this PR and check #8.

Sure, will do.
